pos_no_cash_bank_statement/models/pos_payment_method.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2021 Akretion France (http://www.akretion.com/)
+# @author: Alexis de Lattre <alexis.delattre@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class PosPaymentMethod(models.Model):
+    _inherit = "pos.payment.method"
+    _check_company_auto = True
+
+    cash_journal_id = fields.Many2one(
+        check_company=True,
```

Comment:
Why do you add this attribute ?

Reply:
Because I feel that it is lacking in the point_of_sale module. In fact, this kind of code is not required for this module, I used it as an opportunity to improve the native code. If can remove it from this module if you prefer.
BTW, _check_company_auto =True and check_company=True are still largely ignored by Odoo/OCA devs, and I feel it's really important since odoo v13 which introduced the possibility to have access to several companies at the same time, which adds a risk of selecting a journal from another company than the pos.payment.method. _check_company_auto =True and check_company=True will auto-add a constraint to have an explicit error when saving (cf the ORM code).

Reply:
Hi ! Thanks for your answer.

1) I think you're right regarding the lack of use of ``check_company``.
2) I think it's not the good place to add it here.

If ``check_company`` are missing in many places, I think it's better to have some very simple modules that adds that. 

Idea by module : 
- ``account_check_company``
- ``pos_check_company``
- ``crm_check_company`` ...

or by model: 
- ``account_journal_check_company`` ...

The good place should be in https://github.com/OCA/multi-company/

What do you think ?

Reply:
I agree with you, it's not the role of this module to fix that kind of issues. So I removed _check_company_auto = True and check_company=True